### ISSUE TITLE
EXPIRES shows AM instead of PM

### ISSUE THREAD

**Issue by faceoffers28** [tomasmcguinness/dotnet-passbook#106]
I'm trying to display Dec 31, 2019 at 10:10 PM as the EXPIRES date and time. The TimeZone is "Central Standard Time" (currently 6 hours off) . If I use DateTime from my Asp.Net Web API, the date and time is Dec 31, 2019 at 4:10 PM, which is 6 hours off. If I add a TimeSpan of 6 to the DateTime, I get Dec 31, 2019 at 10:10 AM as the EXPIRES date and time. For some reason, Apple Wallet seems to be incorrectly displaying the time as AM instead of PM.

Here is my code.

`generatorRequest.AddAuxiliaryField(new DateField("expires", "EXPIRES", FieldDateTimeStyle.PKDateStyleMedium, FieldDateTimeStyle.PKDateStyleShort, expiryDate.Add(offset)));
`

I assume that this is wrong as well.

`generatorRequest.ExpirationDate = expiryDate.Add(offset);`

Any help is much appreciated! Thanks!

**Comment by tomasmcguinness** [comment]
Hi.

Adding a TimeSpan,  changes the time, it doesn't account for timezones.

How are you creating the expiryDate object?

**Comment by faceoffers28** [comment]
I see. I'm not doing anything special. I'm working with a nullable DateTime object. The DateTime is stored in my server as UTC. Here is the actual entry.

2019-12-31 22:10:00.000

Here is the code.

var expiryDate = offer.ExpiryDate.HasValue ? offer.ExpiryDate.Value : System.DateTime.Now;

Is there a way to specify AM or PM?

Thanks!

**Comment by tomasmcguinness** [comment]
iOS Wallet converts times actually into the time of the iOS device. In your case, it's correctly converting 22:10, to 16:10 in your first example, accounting for the six hour time difference. 

Adding six hours to your UTC expiryDate should change it to 04:00 on the 1st of Jan, which would then cause Pass to display the 22:10 time correctly in CST.

Feels like something else is at play here.

Can you confirm the JSON being generated contains the correct value in the auxiliary field i.e. has no timezone offset? Feel free to email me the pkpass file and I can take a look.

**Comment by faceoffers28** [comment]
I finally got it to work, but the solution seems odd to me. I had to use the IgnoresTimeZone = true parameter. 

`generatorRequest.AddAuxiliaryField(new DateField("expires", "EXPIRES ON", FieldDateTimeStyle.PKDateStyleMedium, FieldDateTimeStyle.PKDateStyleShort, expiryDate) { IgnoresTimeZone = true });`

The expiryDate (stored in UTC) is coming directly from the database and I'm not using any timezone or timezone offset to convert it.

**Comment by tomasmcguinness** [comment]
Glad this has resolved your issue.

Would you mind checking one thing for me? When you create the DateTime object from the database, does it have a "Kind" of UTC, Local or Unspecified?

**Comment by faceoffers28** [comment]
Honestly, I think I got confused. When I set the ExpirationDate DateTime, that's a specific point in time. When I set the Created/Updated DateTime on a record, that's set to DateTime.UtcNow. Since I'm only concerned about the ExpirationDate, as it relates to passes, there is no need to correct/adjust the DateTime per timezone. Does that make sense? I only need to adjust for timezone when I use DateTime.UtcNow.
